core/src/kvs/tx.rs

```diff
@@ -84,6 +85,108 @@ where
 	pub values: Vec<(Key, Val)>,
 }
 
+pub trait Transactable {
```

Comment:
I know [traits do not have a clearly defined naming convention in Rust](https://github.com/rust-lang/api-guidelines/discussions/28) but this one is very rare. Reading the implementations above felt off because of this. Have you considered naming this just `Transaction`?

Reply:
Unfortunately Transaction is already named by the struct and most of the inner implementations. An alternative would be `TransactionImplementation` but at this point it is not only long, it also sounds like RAS syndrome in terms of meaning. So I opted for `Transactable` which is less verbose and less redundant.

Reply:
That can be worked around by defining one of those types in a separate module. Have you considered moving the trait to `kvs/mod.rs`? You may need to fix some imports afterwards but that should work.

Reply:
Done. While `kvs::Transaction` is not okay because it will glob-import `Transaction` from `tx`, and many other modules are using this as reference. So I moved `Transactable` to `kvs/api.rs` where API related stuff should stay on, and renamed the trait to `Transaction`, available for reference as `crate::kvs::api::Transaction`.